openpype/hosts/maya/plugins/publish/validate_mesh_ngons.py

```diff
@@ -0,0 +1,53 @@
+from maya import cmds
+
+import pyblish.api
+import openpype.api
+import openpype.hosts.maya.api.action
+from avalon import maya
+from openpype.hosts.maya.api import lib
+
+
+def polyConstraint(objects, *args, **kwargs):
+    kwargs.pop('mode', None)
+
+    with lib.no_undo(flush=False):
+        with maya.maintained_selection():
+            with lib.reset_polySelectConstraint():
+                cmds.select(objects, r=1, noExpand=True)
+                # Acting as 'polyCleanupArgList' for n-sided polygon selection
+                cmds.polySelectConstraint(*args, mode=3, **kwargs)
+                result = cmds.ls(selection=True)
+                cmds.select(clear=True)
+
+    return result
```

Comment:
Should this be removed and use `openpype.hosts.maya.api.lib.polyConstraint` instead?

It's using a different mode on `cmds.polySelectConstraint` which we could expose in the lib. But it shouldn't make any difference since both act on the necessary selection.

Also, you might want to update that function in the lib to avoid some slow calls when current tool in Maya is set to move, rotate or scale tool. Here's [our update](https://github.com/Colorbleed/colorbleed-config/commit/559ea17f995d8ff34662710bf10140e8fac39b07) that sets the select tool during the context to avoid it. The slow down was mostly noticable on e.g. vertex selections of full meshes, which was used for other polyConstraint queries which sometimes makes [e.g. this validation super slow](https://github.com/pypeclub/OpenPype/blob/develop/openpype/hosts/maya/plugins/publish/validate_mesh_non_zero_edge.py) on high polycount meshes.